pkg/catalogv2/system/system_test.go

```diff
@@ -38,3 +41,295 @@ func TestGetIntervalOrDefault(t *testing.T) {
+
+			expectedInstalled: false,
+			expectedVersion:   "1.0.0",
+			expectedValues:    newValues,
+			expectedErr:       false,
+		},
+		{
+			name:          "higher min and even higher latest",
+			latestVersion: "1.2.0",
+			minVersion:    "1.1.0",
+			desiredValues: standardValues,
+
+			expectedInstalled: false,
+			expectedVersion:   "1.2.0",
+			expectedValues:    standardValues,
+			expectedErr:       false,
+		},
+		{
+			name:          "higher min and even higher latest with values changed",
+			latestVersion: "1.2.0",
+			minVersion:    "1.1.0",
+			desiredValues: newValues,
+
+			expectedInstalled: false,
+			expectedVersion:   "1.2.0",
+			expectedValues:    newValues,
+			expectedErr:       false,
+		},
+		{
+			name:          "both min and latest are higher",
+			latestVersion: "1.2.0",
+			minVersion:    "1.2.0",
+			desiredValues: standardValues,
+
+			expectedInstalled: false,
+			expectedVersion:   "1.2.0",
+			expectedValues:    standardValues,
+			expectedErr:       false,
+		},
+		{
+			name:          "both min and latest are higher but values changed",
+			latestVersion: "1.2.0",
+			minVersion:    "1.2.0",
+			desiredValues: newValues,
+
+			expectedInstalled: false,
+			expectedVersion:   "1.2.0",
+			expectedValues:    newValues,
+			expectedErr:       false,
+		},
+		{
+			name:          "latest is higher but min is lower than current",
+			latestVersion: "1.1.0",
+			minVersion:    "0.9.0",
+			desiredValues: standardValues,
+
+			expectedInstalled: true,
+			expectedVersion:   "",
+			expectedValues:    nil,
+			expectedErr:       false,
+		},
+		{
+			name:          "latest is higher but min is lower than current but values changed",
+			latestVersion: "1.1.0",
+			minVersion:    "0.9.0",
+			desiredValues: newValues,
+
+			expectedInstalled: false,
+			expectedVersion:   "1.0.0",
+			expectedValues:    newValues,
+			expectedErr:       false,
+		},
+		{
+			name:          "min version is higher but latest is lower than current",
+			latestVersion: "0.9.0",
+			minVersion:    "2.0.0",
+			desiredValues: standardValues,
+
+			expectedInstalled: false,
+			expectedVersion:   "",
+			expectedValues:    nil,
+			expectedErr:       true,
+		},
+		{
+			name:          "invalid latest version",
+			latestVersion: "pug",
+			minVersion:    "0.9.0",
+			desiredValues: standardValues,
+
+			expectedInstalled: false,
+			expectedVersion:   "",
+			expectedValues:    nil,
+			expectedErr:       true,
+		},
+		{
+			name:          "invalid min version",
+			latestVersion: "1.3.0",
+			minVersion:    "pug",
+			desiredValues: standardValues,
+
+			expectedInstalled: false,
+			expectedVersion:   "",
+			expectedValues:    nil,
+			expectedErr:       true,
+		},
+		{
+			name:          "min and latest are both unset",
+			latestVersion: "",
+			minVersion:    "",
+			desiredValues: standardValues,
+
+			expectedInstalled: false,
+			expectedVersion:   "",
+			expectedValues:    nil,
+			expectedErr:       true,
+		},
```

Comment:
Can you add these test cases
``` golang
		{
			name:          "no min version latest version is greater",
			latestVersion: "2.0.0",
			minVersion:    "",
			desiredValues: standardValues,

			expectedInstalled: false,
			expectedVersion:   "2.0.0",
			expectedValues:    standardValues,
			expectedErr:       false,
		},
		{
			name:          "no min version latest version equals current",
			latestVersion: "1.0.0",
			minVersion:    "",
			desiredValues: standardValues,

			expectedInstalled: true,
			expectedVersion:   "",
			expectedValues:    nil,
			expectedErr:       false,
		},
		{
			name:          "no min version latest version equals current values differ",
			latestVersion: "1.0.0",
			minVersion:    "",
			desiredValues: newValues,

			expectedInstalled: false,
			expectedVersion:   "1.0.0",
			expectedValues:    newValues,
			expectedErr:       false,
		},
```

Reply:
I already have these cases in a separate test: `TestIsInstalledExactVersion`.